login/src/main/java/com/fightpandemics/login/ui/signup/SignUpEmailFragment.kt

```diff
@@ -70,53 +70,41 @@ class SignUpEmailFragment : BaseFragment() {
-                            Timber.e("Email verification ${signupResponse.emailVerified}")
+        loginViewModel.signup.observe(
+            viewLifecycleOwner,
+            { signupResponse ->
+                cl_btn_join.isEnabled = true
+                when {
+                    signupResponse.isError -> {
+                        Toast.makeText(
+                            requireContext(),
+                            "Oops something wrong please try again later: " + signupResponse.error,
+                            Toast.LENGTH_SHORT
+                        ).show()
+                    }
+                    else -> {
+                        when {
+                            signupResponse.emailVerified -> {
+                                displayorHideView(View.VISIBLE, cl_btn_join)
+                                Timber.e("Email verification ${signupResponse.emailVerified}")
 
-                            // TODO 9 - Fix this hardcoded string
-                            val PACKAGE_NAME = "com.fightpandemics"
-                            val intent = Intent().setClassName(
-                                PACKAGE_NAME,
-                                "$PACKAGE_NAME.ui.MainActivity"
-                            )
-                            startActivity(intent).apply { requireActivity().finish() }
-                        }
-                        !signupResponse.emailVerified -> {
-                            findNavController().navigate(R.id.action_signupEmailFragment_to_verifyEmailFragment)
+                                val intent = Intent().setClassName(
+                                    packageName,
+                                    "$packageName.ui.MainActivity"
+                                )
+                                startActivity(intent).apply { requireActivity().finish() }
+                            }
+                            !signupResponse.emailVerified -> {
+                                findNavController().navigate(R.id.action_signupEmailFragment_to_verifyEmailFragment)
+                            }
                         }
                     }
                 }
             }
-        })
+        )
     }
 
-    private fun checkLayoutSignUpButton() {
-//        if (validEmail && validPassword && validRePassword) {
-//            val filter: ColorFilter = PorterDuffColorFilter(
-//                resources.getColor(R.color.colorPrimary),
-//                PorterDuff.Mode.SRC_ATOP
-//            )
-//            cl_btn_join.background.colorFilter = filter
-//        } else {
-//            val filter: ColorFilter = PorterDuffColorFilter(
-//                resources.getColor(R.color.color_button_disabled),
-//                PorterDuff.Mode.SRC_ATOP
-//            )
-//            cl_btn_join.background.colorFilter = filter
-//        }
+    companion object {
+        private const val packageName = "com.fightpandemics"
```

Comment:
Should such a constant really live here? Might be easy to miss if the name changes

Reply:
I agee is easy to miss changes, but can not see a better approach, any suggestions?

Reply:
Not sure, could it live somewhere toplevel such that it can be used by all modules?

Reply:
I tried solving this problem by using navigation actions, but then I had the problem of not being able to reference the splash screen from the login module since the splash screen is part of the app module. Not being able to reference the splash screen makes it not possible for me to pop the navigation stack to prevent the user from using the back button to navigate back from the main feed to the onboarding screens. 
To have a reference to the splash screen fragment from the app module, I attempted to add the app module as a dependency for the login module, but that causes a circular dependency error.
I am unsure of how to continue now regarding this refactoring.

Reply:
I will make this a separate issue